packages/slice-machine/src/features/settings/git/GitRepositoryConnectDialog.tsx

```diff
@@ -0,0 +1,86 @@
+import {
+  Box,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogHeader,
+  FormInput,
+  Text,
+} from "@prismicio/editor-ui";
+import type { GitRepo, GitRepoSpecifier } from "@slicemachine/manager";
+import { Formik } from "formik";
+import type { FC, ReactNode } from "react";
+import { toast } from "react-toastify";
+
+import { gitProviderToConfig } from "@src/features/settings/git/GitProvider";
+import { useWriteAPIToken } from "@src/features/settings/git/useWriteAPIToken";
+
+type GitRepositoryConnectDialogProps = {
+  linkRepo: (repo: GitRepoSpecifier) => Promise<void>;
+  repo: GitRepo;
+  trigger: ReactNode;
+};
+
+export const GitRepositoryConnectDialog: FC<
+  GitRepositoryConnectDialogProps
+> = ({ linkRepo, repo, trigger }) => {
+  const providerConfig = gitProviderToConfig[repo.provider];
+  const { updateToken } = useWriteAPIToken({ git: repo });
+  return (
+    <Dialog size={{ width: 448, height: "auto" }} trigger={trigger}>
+      <DialogHeader title="Prismic write API token required" />
+      <DialogContent>
+        <Formik
+          initialValues={{ writeAPIToken: "" }}
+          validate={(values) => {
+            if (values.writeAPIToken.length === 0) {
+              return { writeAPIToken: "Cannot be empty" };
+            }
+          }}
+          onSubmit={async (values) => {
+            try {
+              await updateToken(values.writeAPIToken);
+              await linkRepo(repo);
```

Comment:
`linkRepo()` should happen before `updateToken()`; updating the token requires that the repository link exists first.